src/MetaModels/MetaModel.php

```diff
@@ -730,27 +730,37 @@ public function getIdsFromFilter($objFilter, $strSortBy = '', $intOffset = 0, $i
             } elseif ('id' === $strSortBy) {
                 asort($arrFilteredIds);
             } elseif (in_array($strSortBy, array('pid', 'tstamp', 'sorting'))) {
-                // Check existing ids.
-                if (array_intersect($arrFilteredIds, $this->existingIds) == $arrFilteredIds) {
-                    return $arrFilteredIds;
+                // Build the right key for the cache.
+                $sortKey = \sprintf('%s-%s', $strSortBy, \strtolower($strSortOrder));
+                // Used the cached ID list, and make a list of wanted ID's with the sorting of the cache.
+                $cacheResult = array_intersect($this->existingIds[$sortKey], $arrFilteredIds);
+                // Check if we have all ID's or if we have one missing, now we are using the order of the MM Filter.
+                if (array_intersect($arrFilteredIds, $cacheResult) == $arrFilteredIds) {
+                    return $cacheResult;
                 }
-                
+
+                // Merge the already known and the new one.
+                $fullIdList = array_merge($this->existingIds[$sortKey], $arrFilteredIds);
```

Comment:
Maybe we have to cast the existingIds to an array:  ``` $fullIdList = array_merge((array) $this->existingIds[$sortKey], $arrFilteredIds); ```

Reply:
fixed